### ISSUE TITLE
Webogram is using an outdated Emoji set. Please update the Emoji set to the latest version and please add the skin tone modifier feature to the Webogram Emoji drawer.

### ISSUE THREAD

**Issue by wp9015362** [zhukov/webogram#760]
@zhukov :

The Emoji set in Webogram is now outdated, see:

http://blog.emojipedia.org/apple-2015-emoji-changelog-ios-os-x

Could you please update the Emoji set to the latest version and could you please add the skin tone modifier feature to the Webogram Emoji drawer?

Regards

**Comment by wp9015362** [comment]
PS:

And please also add the new Emoji categories to the Telegram for Android Emoji drawer, which are also described over there:

> http://blog.emojipedia.org/apple-2015-emoji-changelog-ios-os-x
>
> 😃 People
> 🌸 Nature
> 🍔 Food & Drink
> 🎉 Celebration
> 🏃 Activity
> 🌇 Travel & Places
> 🔣 Objects & Symbols

Regards

**Comment by sergioad** [comment]
Or maybe can you use the Twitter emojis?

**Comment by wp9015362** [comment]
> @sergioad wrote:
> 
> Or maybe can you use the Twitter emojis?

The open source Twitter "Twemoji" Emoji set:

https://blog.twitter.com/2014/open-sourcing-twitter-emoji-for-everyone
https://github.com/twitter/twemoji

indeed would be quite smiliar, see the following comparison for example:

http://mroth.github.io/emojicompare/

This comparison is showing an outdated version of the Twitter "Twemoji" Emoji set though.

Over the last couple of days the Twitter "Twemoji" Emoji set already has received some nice improvements, see:

https://github.com/twitter/twemoji/commits/gh-pages
https://github.com/twitter/twemoji/issues
https://github.com/twitter/twemoji/issues?q=is%3Aissue+is%3Aclosed

So, the updated set looks better now. They still need to create PNGs from the updated SVGs though, see:

https://github.com/twitter/twemoji/issues/60

> @robin0van0der0v wrote:
> 
> The Twitter emoji set also doesn't have the colored emoji yet.

They seem to be working on it though, see:

https://github.com/twitter/twemoji/issues/48

Regards

**Comment by wp9015362** [comment]
@zhukov :

WhatsApp (incl. WhatsApp Web) uses the updated set now and even has the the skin tone modifier feature for it's built-in Emoji drawer now, see the following article and screenshots for example:

http://www.androidpolice.com/2015/06/30/live-long-and-prosper-whatsapp-adds-skin-tone-emojis-and-spock-vulcan-salute-on-android/

Will Webogram also get it anytime soon?

Regards

**Comment by mwld** [comment]
More emojis have recently been added with iOS 9.1 update: http://blog.emojipedia.org/ios-9-1-emoji-changelog/

**Comment by stefan-schweiger** [comment]
Also WhatsApp has already integrated those. I know only the sky is the limit with stickers, but it would be nice to have the new Emojis and not be behind WhatsApp on this one.

**Comment by bluezed** [comment]
Looks like the Telegram Android App supports the new Emojis now too:
![Telegram App](http://s19.postimg.org/82ji985er/App.png)

And this is how it looks in the Webogram:
![Webogram](http://s19.postimg.org/qgu1d7hpf/Web_App.png)

**Comment by julianxhokaxhiu** [comment]
Is there something planned about this?

**Comment by theus** [comment]
Telegram Desktop has added all emojis in [an update](https://github.com/telegramdesktop/tdesktop/releases/tag/v0.9.15) last week

**Comment by tinchou** [comment]
Maybe we can take advantage of the work done in the mobile app? e.g.: https://github.com/DrKLO/Telegram/blob/496c336d5f80947ebce64087cc9740b3d28ee99d/TMessagesProj/src/main/java/org/telegram/messenger/EmojiData.java

**Comment by jusore** [comment]
Will be this bug will fixed this year?

**Comment by Akronix** [comment]
I'm trying to help with this to at least add the new emojis that are in every official telegram client except for webogram, but I can't find where exactly the images are loaded from.
Could you please give us a clue @zhukov?

**Comment by gorghino** [comment]
Nearly April 2017. Another bug birthday :(

**Comment by Ryuno-Ki** [comment]
Patches welcome? Look at the [contributors graph](https://github.com/zhukov/webogram/graphs/contributors?from=2014-05-09&to=2014-10-27&type=c)! The number is quite low!

**Comment by bluezed** [comment]
Looks like something has changed, maybe with the Telegram API? I suddenly see Emojis that weren't supported by webogram though they look slightly different to the "normal" emojis.

See here:
![Emoji screenshot](https://image.ibb.co/dF8tQa/Emojis1.png)

And this is how the HTML looks:
![Source screenshot](https://image.ibb.co/n7ThJv/Emojis2.png)

Only thing missing now is the ability to actively use those emojis in webogram, not just display them ;)

**Comment by tinchou** [comment]
It's probably the browser's native support for emojis.

**Comment by stek29** [comment]
@bluezed Webogram thinks that those symbols aren't emoji and doesn't replace them, so system can render them. It's easy to notice how different they are.

(btw, what is that screenshot from? What browser does allow to theme devtools?)

**Comment by bluezed** [comment]
Thanks for the info.

That's Firefox 52.0.2 (32bit). Not changed any themes though.



Am 01.04.2017 4:26 vorm. schrieb "Viktor Oreshkin" <notifications@github.com

**Comment by Ryuno-Ki** [comment]
Firefox does allow [theming the developer tools](https://developer.mozilla.org/en-US/docs/Tools/Settings#Themes) since a while. The latest theme was Firebug-like.

**Comment by jordansteeves** [comment]
please implement this!

**Comment by codefuhrer** [comment]
Bump. There are a lot of missing emojis.

**Comment by stormyqq** [comment]
it has been so long hope the emojis will be updated in the desktop version soon.

**Comment by stek29** [comment]
@stormyqq emoji in TDesktop were updated months ago.

**Comment by stormyqq** [comment]
I am not sure it's updated or not, but the current emojis ain't that good looking.  I prefer if it's something like this https://emoji.codes/ 
can't we at least replace the existing emojis in the telegram?

**Comment by stefan-schweiger** [comment]
Replacing them in their entirety is just as tedious as just getting the new ones to work, which already no one seems to cares about...
Also I'm not a fan of using another emoji style because it makes them inconsistent to other systems (emoji one is not bad, but still sometimes you don't convey the same meaning when they look different)

**Comment by stormyqq** [comment]
well I care about. the existing emojis are so small, and old looking, I like how telegram has this feature which allows us to create our own stickers, I guess if they allow to create our own emojis like that, then that would be great for everyone.

**Comment by stefan-schweiger** [comment]
@stormyqq good for you, if you care enough implement it yourself. it's an open source project after all. what I meant no one of the active development team seems to care to do the work, so either change it your self or live with the status quo or just use the desktop app ;)

**Comment by stormyqq** [comment]
Now you sound rude here, isn't this section for requesting new features? why you argue with people who request new features? I don't want to argue with anyone, if they don't like they don't have to implement it.

**Comment by stefan-schweiger** [comment]
@stormyqq look this issue is open for over two years now, the core team clearly isn't going to fix it, thats all I'm saying. So as long as someone doesn't decide to do it him/herself this don't expect it to change anytime soon.

**Comment by Ryuno-Ki** [comment]
The core team consusts of 1 person. I can imagine, emoji aren't prio 1.

Contributions welcome :)

**Comment by stek29** [comment]
@Ryuno-Ki And that person is doing a lot of other stuff in Telegram :)

**Comment by jvschiavo** [comment]
Does anyone know to do it? I don't know how to do that kind of thing. But I really wanted to help. :(

**Comment by jvschiavo** [comment]
Hey @stek29, could you work into this? I really want to help but I need to be guided since I don't know about this. =(

**Comment by stek29** [comment]
@jvschiavo I'd love to help, but I have no idea about how exactly tgweb works with emojis.
I'd suggest you to port emojione libs and remove webogram's stuff tho.

**Comment by jvschiavo** [comment]
@stek29 you should forget emojione, they are not free anymore.

**Comment by stek29** [comment]
@jvschiavo Their JS Library is. I'm not talking about porting emojione emoji, I suggest to port emojione JS library.

**Comment by Discostu36** [comment]
Just wanted to bump this because Apple released new Emojis with iOS 11.1 on October 31st. The number of Emojis that Telegram Web does not support is growing and growing. This is really an important feature! How can I react adequately to  messages if I can't see 1/3 of the emojis they contain?

**Comment by gorghino** [comment]
I stopped using Telegram Web because of this bug. Telegram desktop client is also waiting for new Emojis

**Comment by Ryuno-Ki** [comment]
Use emoticons :|

**Comment by gorghino** [comment]
Welcome to 1980

**Comment by stek29** [comment]
@gorghino 🚫♂️N0️⃣👋👋 WELCOME🤙😂😉TO2️⃣0️⃣1️⃣8️⃣

**Comment by ingria** [comment]
:man_facepalming:

**Comment by Ryuno-Ki** [comment]
![screenshot_20171128-222810](https://user-images.githubusercontent.com/3097194/33345108-79d52d58-d48b-11e7-95f3-8b8cf307fab8.png)

Emojis are inherently flawed. I was serious regarding the emoticons. No argue on skin colour and so on ;-)

**Comment by jvschiavo** [comment]
![screenshot_20171201-160628](https://user-images.githubusercontent.com/20977075/33496535-bcefcada-d6b1-11e7-86de-91dcc1ac49b5.png)

I have no problems on a **decent** Android phone. :)

**Comment by Ryuno-Ki** [comment]
Fairphone 2 without Google here. Maybe I am missing some fonts. However, this could happen to anybody, so I would recommend against Emojis. Maybe we can find common ground in having them written as `:thumbs_up:` and replaced by the emoji if supported (however support would be determined).

**Comment by doktormerlin** [comment]
2018 now and still iOS 9 emojis are used, without skin tone modifier. I think the emoji set should be updated at least, skin tone modifier being no priority here.

**Comment by ingria** [comment]
Emojis are now hosted on telegram.org. No need to find the lib:

https://telegram.org/img/emoji/40/F09F948D.png
https://telegram.org/img/emoji/40/F09F9A80.png
https://telegram.org/img/emoji/40/F09F9281E2808DE29982.png

**Comment by Carighan** [comment]
Another quarter has passed with no update. I wanted to add that if it is a technical limitation which makes it difficult to use the new iOS emoji set, maybe switching to google's emoji or EmojiOne could alleviate it? Whatever can be updated the fastest.

**Comment by ingria** [comment]
Well, I've managed to implement the full emoji support in my fork [demo](https://botcontrol.ru/web). The only problem is that I couldn't find the authentic Apple emoji font/images/sprites/whatever, so I had to use Emojione resources (which, I suppose, are unsuitable for the official Telegram Web app).



So, if anyone helps me to find an up-to-date Apple emoji images, I can PR to this repo.

Also I suggest changing the emoji size from various sizes to fixed one: this will 1) reduce the size of the the sprite, 2) improve performance (background-image resizing is very resource-consuming).

**Comment by ingria** [comment]
Emojis are ideograms. Meaning that they are as much bogus additions to Unicode, as chinese hieroglyphs or any other logogram (hint: not bogus at all).

**Comment by Carighan** [comment]
@ingria : Emojione seems more sensible to me anyhow, Telegram is not just an iOS/OSX client.

**Comment by stek29** [comment]
@Carighan Exactly: Telegram is not just a web client, and that's why it can't use different emoji set.

@ingria AFAIK Telegram actually pays apple licensing fees for emoji, so I'm not sure if PR could be accepted. Also, You can use https://github.com/telegramdesktop/tdesktop/blob/dev/Telegram/Resources/art/emoji_250x.webp (which is somewhat outdated too, I think), or probably pull out resources from Android app.

**Comment by ingria** [comment]
I have a script that creates a sprites with any emoji images and generates a JS/SCSS code. It doesn't care what images to use — Apple's, Emojione's or whatever. So I'm ready to PR any moment, as soon as I have full emoji set (and info about licensing if appropriate).

@Carighan yeah, I agree. I also like their style more than Apple's. But Telegram team aims to consistency across the clients and OSes, so they (most likely) won't accept anything except Apple emojis :man_shrugging:

**Comment by StefanBrand** [comment]
@ingria Thanks a lot! I get that https://botcontrol.ru/web is possible only because Telegram's API is open per se, but the official Telegram Web App could do better in regards to emojis.

**Comment by flying-sheep** [comment]
WhatsApp semi-recently introduced an emoji set that’s similar to Apple’s, but not a copy, and with less of that hideous gloss.

Maybe Telegram could investigate if those are available without fees.

But anyway, the webogram emoji picker is hilariously outdated, and the only thing uglier than Apple emoji is seeing Apple emoji mixed with my system font’s emoji (when my conversation partner uses a newer one not available in webogram)

**Comment by ingria** [comment]
No point to make a PR because this project is dead. The new one is being developed [here](https://github.com/evgeny-nadymov/telegram-react), ([demo](https://evgeny-nadymov.github.io/telegram-react/)).

**Comment by TLuigi003** [comment]
That "new one" is not official.
If telegram has an oficial client, which is the one most people are going to use, we should try to fix it

He seems to have deleted his fork?

**Comment by Carighan** [comment]
> Edit: Ok, I see that the domain of @ingria's fork is https://web.tlgrm.eu now. I'll check that out.
> Edit2: web.tlgrm.eu works at University of Vienna. 😉

Is that actually official? Because the official page links to the old webogram.

**Comment by StefanBrand** [comment]
No, it's not official. But since webogram is open source and the "official" webogram hasn't updated the emoji set, I don't see a problem in using a fork.

**Comment by flying-sheep** [comment]
Where’s the code for the fork? @ingria?

**Comment by KevDoy** [comment]
Any movement on this?

**Comment by ricardofiorani** [comment]
Any update on this ?
